src/main/drivers/stm32/dshot_bitbang.c

```diff
@@ -500,11 +509,22 @@ static bool bbUpdateStart(void)
 #ifdef USE_DSHOT_TELEMETRY_STATS
         const timeMs_t currentTimeMs = millis();
 #endif
-        timeUs_t currentUs = micros();
 
-        // don't send while telemetry frames might still be incoming
-        if (cmpTimeUs(currentUs, lastSendUs) < (timeDelta_t)(40 + 2 * dshotFrameUs)) {
-            return false;
+        // Wait for telemetry reception to complete before decode
+        bool telemetryPending;
+        bool telemetryWait = false;
+
+        do {
+            telemetryPending = false;
+            for (int i = 0; i < usedMotorPorts; i++) {
+                telemetryPending |= bbPorts[i].telemetryPending;
+            }
+
+            telemetryWait |= telemetryPending;
+        } while (telemetryPending);
```

Comment:
Can this hang up in case telemetryPending never goes to ´0´ in time?

Reply:
I don't believe that will ever happen as its internally timed and not dependant on any external device.

Reply:
Timeout added.